interfaces.go

```diff
@@ -0,0 +1,171 @@
+package linodego
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/linode/linodego/internal/parseabletime"
+)
+
+type LinodeInterface struct {
+	ID           int                    `json:"id"`
+	MACAddress   string                 `json:"mac_address"`
+	Created      *time.Time             `json:"-"`
+	Updated      *time.Time             `json:"-"`
+	DefaultRoute InterfacesDefaultRoute `json:"default_route"`
+	Version      int                    `json:"version"`
+	VPC          InterfacesVPC          `json:"vpc"`
+	Public       InterfacesPublic       `json:"public"`
+	VLAN         InterfacesVLAN         `json:"vlan"`
+}
+
+type InterfacesDefaultRoute struct {
+	IPv4 bool `json:"ipv4,omitempty"`
+	IPv6 bool `json:"ipv6,omitempty"`
+}
+
+type InterfacesVPC struct {
+	SubnetID int             `json:"subnet_id"`
+	IPv4     *InterfacesIPv4 `json:"ipv4,omitempty"`
+}
```

Comment:
```suggestion
type InterfacesVPC struct {
	VPCID    int `json:"vpc_id"`
	SubnetID int             `json:"subnet_id"`
	IPv4     *InterfacesIPv4 `json:"ipv4,omitempty"`
}
```
VPC ID is missing here

Reply:
Should we separate return and post objects for these interfaces? `vpc_id` will only appear in the returning object...